File: 26-50/027.cpp
// BY {prateekkol21} copyright comment 

#include <stdio.h>

int isPrime(int n){
  int i;
  if (n<=0)
    return 0;
  if (n==2)
    return 1;
  for (i=2;i<=n/2;i++)
    if(n%i==0)
      return 0;
  return 1;
}

int main(){
  int a,b,n,max,supermax,maxa,maxb;
  supermax=0;
    int N;
    scanf("%d",&N);
  for (a=-999;a<N;a++){
    for (b=-999;b<N;b++){
      max=0;
      for (n=0;n<N;n++){
        if(!isPrime((n*n)+(a*n)+b))
          break;
        max++;
      }
      if (max>supermax){
        supermax=max;
        maxa=a;
        maxb=b;
      }
    }
  }  
  printf("%d %d\n",maxa,maxb);      
return 0;
}

File: 26-50/027.java
public class 027 {
	public static void main(String[] args) {
		int maxNum = 0;
		int maxA = 0;
		int maxB = 0;
		for(int a = -1000; a <= 1000; a++) {
			for(int b = -1000; b <= 1000; b++) {
				int consecutivePrimes = numConsecutivePrimes(a, b);
				if(consecutivePrimes > maxNum) {
					maxNum = consecutivePrimes;
					maxA = a;
					maxB = b;
				}
			}
		}
		System.out.println(maxA * maxB);
	}
	public static int numConsecutivePrimes(int a, int b) {
		int i = 0;
		while(isPrime(i * i + a * i + b)) {
			i++;
		}
		return i;
	}
	public static boolean isPrime(int n) {
		if(n == 2) {
			return true;
		}
		if(n <= 1 || n % 2 == 0) {
			return false;
		}
		for(int i = 3; i * i <= n; i += 2) {
			if(n % i == 0) {
				return false;
			}
		}
		return true;
	}
}
